scripts/armadillo-setup.sh

```diff
@@ -0,0 +1,298 @@
+          ;;
+        --oidc_clientsecret)
+          OIDC_CLIENTSECRET=$2
+          shift 2
+          ;;
+        --cleanup)
+          ARMADILLO_CLEANUP=1
+          shift          ;;
+
+        -h | --help)
+          parameters_help
+          exit 0
+          ;;
+        -*)
+          parameters_help
+          exit 0
+          ;;
+        *)
+        break;
+        esac
+        
+    done
+    if [ ! "$ARMADILLO_CLEANUP" ]; then
+      if [ ! "$ARMADILLO_DOMAIN" ] || [ ! "$ARMADILLO_ADMIN_PW" ]; then
+        echo "Arguments --domain --admin-password must be provided"
+        echo "U need an host or domain to use Armadillo: Example: cohort.armadillo.organisation.com"
+        echo "Also for security reasons you must provide an secure admin password"
+        parameters_help; 
+        exit 1;
+      fi
+    fi
+    if [ "$ARMADILLO_OIDC_ENABLED" ]; then
+      if [ ! "$OIDC_CLIENTID" ] || [ ! "$OIDC_ISSUER_URL" ] || [ ! "$OIDC_CLIENTSECRET" ]; then
+        echo "OIDC Option called but mandatory config items are missing --oidc_url <issuer_url> --oidc_clientid <client_id> --oidc_clientsecret <secret> "
+        exit 1;
+      fi
+    fi
+
+
+}
+
+
+setup_environment() {
+    mkdir -p $ARMADILLO_PATH/application
+    mkdir -p $ARMADILLO_PATH/services
+    mkdir -p "$ARMADILLO_LOG_PATH"
+    mkdir -p "$ARMADILLO_CFG_PATH"
+    mkdir -p "$ARMADILLO_DATADIR"
+    useradd -rs /bin/false "$ARMADILLO_SYS_USER"
+    chgrp -R "$ARMADILLO_SYS_USER" "$ARMADILLO_PATH"
+    chgrp -R "$ARMADILLO_SYS_USER" "$ARMADILLO_CFG_PATH"
+    chgrp -R "$ARMADILLO_SYS_USER" "$ARMADILLO_LOG_PATH"
+    chgrp -R "$ARMADILLO_SYS_USER" "$ARMADILLO_DATADIR"
+
+    chmod g+rw "$ARMADILLO_LOG_PATH"
+    chmod g+rw "$ARMADILLO_DATADIR"
+    usermod -aG docker "$ARMADILLO_SYS_USER"
+    echo "Environment is beeing setup correctly"     
+}
+
+setup_systemd() {
+    # Set the systemd version. In future download from github? Example:
+    #curl -fsSL -o /etc/systemd/system/armadillo.service https://github.com/molgenis-service-armadillo/raw/branch/master/dist/etc/systemd/system/armadillo.service
+    cat  > /etc/systemd/system/armadillo.service << EOF
+[Unit]
+Description=DataSHIELD Armadillo 3
+After=syslog.target
+
+[Service]
+User=$ARMADILLO_SYS_USER
+Environment=SPRING_PROFILES_ACTIVE=$ARMADILLO_PROFILE
+Environment=SPRING_CONFIG_LOCATION=$ARMADILLO_CFG_PATH/application.yml
+WorkingDirectory=$ARMADILLO_PATH
+ExecStart=java -jar $ARMADILLO_PATH/application/armadillo.jar
+StandardOutput=append:$ARMADILLO_LOG_PATH/armadillo.log
+StandardError=append:$ARMADILLO_LOG_PATH/error.log
+Type=simple
+Restart=on-failure
+RestartSec=10
+
+[Install]
+WantedBy=multi-user.target
+
+EOF
+
+systemctl daemon-reload
+echo "Armadillo Installed under systemd"
+}
+
+setup_armadillo_config() {
+  if [ "$ARMADILLO_OIDC_ENABLED" ]; then
+    wget -q -O $ARMADILLO_CFG_PATH/application.yml https://raw.githubusercontent.com/molgenis/molgenis-service-armadillo/scripts/conf/application-oidc.yml
```

Comment:
Do you think we should pin these URLs to a specific version somehow? Now the script will always get the latest version from the dev/master branch. We might make changes there before a new version has been released, potentially breaking older installations. Another strategy is to always make this installation script backwards compatible. Maybe not something we need to fix ASAP, but something to keep in mind for the future.

Reply:
Then we should add both script and yaml to the releases. Then you can get it there. Or use tags and get it from github repo of a particular version. Or more easy, use simply the tags.

Reply:
BTW: I plan to change Armadillo repo to behave similar to emx2, which means any merge into the master is also a release. In that case master is always latest release and then this is generally not a problem.

Reply:
At my opinion is that configuration is an one time only process at initial setup. After installation the config files shouldn't be updated anymore (not automatically). So when you upgrade armadillo and add config features it should not be mandatory. (Or it has to be a major release, where you instruct users to update their config)